### PR TITLE
ENG-212: Rsa key functionality

### PR DESCRIPTION
**Pull request by amiller68**
## Summary
- Made the Rsa Exchange key implementation (used by Fission for testing) a crate export
- upped the default bit size to 3072
- Added functionality for Rsa Public Keys st we can
  - fingerprint the SPKI bytes of the key using sha1
  - import and export to and from pem files
  - import from SPKI bytes
- Added functionality for Rsa Private Keys st we can
  - import and export to and from pem files
- Added tests for same

### CODE REVIEW

File: wnfs/src/private/keys/exchange.rs [hunk 1]
```diff
@@ -138,6 +203,80 @@ mod test {
     }
 
     #[async_std::test]
+    async fn test_rsa_priv_key_from_pem_file() {
```
Comment (recursivepaws) [inline wnfs/src/private/keys/exchange.rs]: nit: you can have your test functions return Results such that you dont need to use as many unwraps. just less cluttered when using ?'s instead

**Review by recursivepaws — COMMENTED** [review]
code looks good, though would like to see my nit comment implemented for test cases

File: wnfs/src/private/keys/exchange.rs (hunk 1 above)
Comment (amiller68) [reply wnfs/src/private/keys/exchange.rs]: consider it done

**Review by sstelfox — CHANGES_REQUESTED** [review]
I looked through this PR and I think all of this belongs in tomb not in a fork for rs-wnfs. We can't upstream any of this and probably don't want to use their test implementation of the RSA keys directly. Those are effectively very thin wrappers over the `rsa` crate key type providing just a simple wrapper and exposing opinionated encrypt and decrypt operations. Replacing that in our code base is fine as it's a very small amount of code.

**Comment by amiller68** [comment]
Superseded by: https://github.com/banyancomputer/tomb/pull/163

**Pull request closed without merging**